.github/workflows/main.yml

```diff
@@ -5,24 +5,31 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+  schedule:
+    # Midnight UTC:
+    - cron: "0 0 * * *"
 
 jobs:
 
   test:
     runs-on: ${{ matrix.os }}
     name: ${{ matrix.os }} - ${{ matrix.python }}
     strategy:
+      fail-fast: false
       matrix:
-        os: [windows-latest, ubuntu-latest]
-        python: [3.9]
+        # Test all supported versions on Ubuntu:
+        os: [ubuntu-latest]
+        python: ["3.7", "3.8", "3.9", "3.10", 3.11-dev]
         include:
-          - os: ubuntu-latest
-            python: 3.6
-          - os: ubuntu-latest
-            python: 3.9
+          # Also test PyPy, macOS, and Windows:
           - os: ubuntu-latest
             python: pypy3
+          - os: macos-latest
+            python: "3.10"
+          # XXX: We should be testing 3.10, but for some reason setuptools can't
+          # seem to find the runner's C compiler for new wheel builds... 
```

Comment:
Maybe it's using an old version of setuptools?

Reply:
Hm, that could be it! Let's find out.

Reply:
Oh, wait. We already upgrade `setuptools` first (see line 36).

In fact, I just remembered that this is where I had seen that error message you've been dealing with this week! I guess my solution was "just use 3.9 for Windows CI":

```
  running build_ext
  building 'dulwich._objects' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

Encountered error while trying to install package.

dulwich

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
Command 'D:\a\pyperformance\pyperformance\venv\cpython3.10-3104833f480c-compat-f6a835d45d46\Scripts\python.exe' -m pip install asgiref==3.3.4 django==3.2.4 pytz==2021.1 sqlparse==0.4.1 tornado==6.1 pyaes==1.6.1 chameleon==3.9.1 mpmath==1.2.1 sympy==1.8 html5lib==1.1 six==1.16.0 webencodings==0.5.1 greenlet==2.0.0a1 sqlalchemy==1.4.19 mako==1.1.4 markupsafe==2.0.1 greenlet==2.0.0a1 sqlalchemy==1.4.19 genshi==0.7.6 six==1.16.0 certifi==2021.5.30 dulwich==0.20.23 urllib3==1.26.5 failed with exit code 1
Error: Process completed with exit code 1.
```

Reply:
Hm, but in my case I'm pretty sure the error was because there were actually compilation errors. (Right now I can't verify because I'm away from my work laptop.)